lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -79,12 +81,18 @@ DartType? getExpectedType(PostfixExpression node) {
         : null;
   }
   // assignment
-  if (parent is AssignmentExpression &&
-      parent.operator.type == TokenType.EQ &&
-      (parent.leftHandSide is! Identifier ||
-          node.operand is! Identifier ||
-          (parent.leftHandSide as Identifier).name !=
-              (node.operand as Identifier).name)) {
+  if (parent is AssignmentExpression && parent.operator.type == TokenType.EQ) {
+    var lhs = parent.leftHandSide;
+    var rhs = node.operand;
+    if (lhs is Identifier && rhs is Identifier && lhs.name == rhs.name) {
+      return null;
+    }
+    if (lhs is SimpleIdentifierImpl) {
+      var scope = lhs.scopeLookupResult?.getter;
```

Comment:
@bwilkerson See why I want to put `resolution` into identifiers? He does not know that the `writeElement` from the parent `AssignmentExpression` should be used here.

Reply:
It's an interesting assumption that `resolution` would have made the API easier to learn. :-)